### ISSUE TITLE
"Unable to embed media" when pasting text and URL to the editor

### ISSUE THREAD

**Issue by reginabally** [wordpress-mobile/gutenberg-mobile#4236]
**Describe the bug**
Originally reported in 4446003-zd-woothemes.

This may not be a bug but an improvement request to the editor when the user is pasting text and URL (in two separate paragraphs) to the editor. The URL will be automatically converted into an Embed block (which is expected since [version 18.5](https://github.com/wordpress-mobile/WordPress-iOS/releases/tag/18.5)) but it shows the "Unable to embed media" error for URL that isn't yet supported in the app. The URL is displayed as it is on the actual site but it is confusing to the user who thought there's an issue inserting the link to the content. 

If we pasted only the URL to an empty Paragraph block, there is a prompt asking if we want to create an embed or link. Would it be possible to show this pop-up as well when pasting text + URL in the editor? Or just show the plain URL for links that are not embeddable in the app's editor?

**To Reproduce**
1. On a note app (e.g. Simplenote), create a note with some text and a URL (not embeddable in the app) in two separate lines/paragraphs. The sample URL in my testing is https://www.bbc.com/news/av/world-us-canada-59229671.
2. Copy the text + URL in the note app.
3. Open the WordPress mobile app, create a new post or page and paste the content to the editor.
4. Notice the URL is converted into an Embed block with the "Unable to embed media" error shown on the block.

**Expected behavior**
The user expects that the URLs will not be automatically converted into the Embed block or if there's an option to disable embed automatically.

**Screen recording**

https://user-images.githubusercontent.com/40906847/141399881-169bfb4e-8e20-4b10-9f8c-1a8e23b5df8b.mp4

**Smartphone (please complete the following information):**
 - Device: iPhone 11
 - OS: iOS 14.8.1
 - WPiOS 18.5

**Comment by jd-alexander** [comment]
Thanks for sharing this detailed report @reginabally 🙇🏾 

> The user expects that the URLs will not be automatically converted into the Embed block or if there's an option to disable embed automatically.

Understood! Currently, this is the behavior of our Paragraph and Embed blocks, where once a link is pasted into a Paragraph block on the web and within the mobile app, we present the Embed options. In this case, we don't support showing the prompt, therefore this will require further exploration.

> Would it be possible to show this pop-up as well when pasting text + URL in the editor? 

This is something we should consider. WDYT @fluiddot ?

For more context, we could show this pop-up for this use case where there is text and a link but things get a bit more tricky if we were to have text and multiple links. I created an issue to track the behavior of multiple links [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/4237
). In that case, we would have to consolidate the behavior so that the prompt is not shown for every pasted URL, but give the user the ability to choose whether they want all links to become Embed blocks or links.

> Or just show the plain URL for links that are not embeddable in the app's editor?

This is a good idea. If we were to implement it, we would be deviating that how the behavior functions on the web. That is the only caveat why we didn't choose this route initially. On the web, if we paste this link this is the result. 

![bbc](https://user-images.githubusercontent.com/1509205/141406333-26cfc659-f2bf-4ac3-b4cf-2c286a89a99c.png)

**Comment by hypest** [comment]
> Or just show the plain URL for links that are not embeddable in the app's editor?

I think that would be a bit confusing since it will not be clear why that link didn't automatically transform as other links. It will need extra UI to denote what went wrong and what the user can do next.

**Comment by hypest** [comment]
> If we pasted only the URL to an empty Paragraph block, there is a prompt asking if we want to create an embed or link. Would it be possible to show this pop-up as well when pasting text + URL in the editor?

I agree that there's a mismatch of behavior here and since we ask to convert when pasting a URL, we should ask when pasting "some test + URL" as well instead if directly converting it.


> **Expected behavior**
>
> The user expects that the URLs will not be automatically converted into the Embed block or if there's an option to disable embed automatically.

I think that it should be safe to go with presenting the question to embed or not instead of automatically not embedding or adding an option. In that sense @reginabally , what do you think about converting this ticket to a "bug" ticket and have it capture the need to introduce the embed-or-not dialog for the test+URL case? By the way, thanks for opening a ticket for the multiple-URLs case @jd-alexander, we could instead re-purpose that ticket into a bug ticket if the consensus is to keep this one as a "don't convert or add a setting" one.

**Comment by iamgabrielma** [comment]
Adding some user feedback, 4446003-zen :

> The best solution I think is to have an option in the app's basic settings which lets each user decide whether all links in all articles should be auto-embedded or
not.

> If some users choose to have auto-embed then let there be an option near each thumbnail to convert it to a plain url link. This option already exists when the auto-embed fails, but there is currently no way to convert successful auto-embedded thumbnails into plain url links.

> Or make it even easier: if one uses a finger on the touchscreen to press (for a second) on the auto-embedded thumbnail, then let the url link be automatically copied and the thumbnail automatically erased, so that one can manually embed the link in a word or in part of a sentence.

**Comment by hypest** [comment]
> If some users choose to have auto-embed then let there be an option near each thumbnail to convert it to a plain url link. This option already exists when the auto-embed fails, but there is currently no way to convert successful auto-embedded thumbnails into plain url links.

Ah, interesting. That's something not currently offered by the mobile nor the web block editor so, that alone can be taken out into a separate enhancement ticket on [the Gutenberg repo](https://github.com/WordPress/gutenberg).

Edit: this [undo to remove Embed issue](https://github.com/WordPress/gutenberg/issues/35335) seems related.

**Comment by reginabally** [comment]
> what do you think about converting this ticket to a "bug" ticket and have it capture the need to introduce the embed-or-not dialog for the test+URL case?

Hi @hypest Yes, it makes sense to me! Thanks!
